### ISSUE TITLE
Migration to type hints

### ISSUE THREAD

**Issue by jawooll** [seisbench/seisbench#130]
With type hinting becoming both standardised and stable within the latest Python versions, maybe we should consider including type hints with the next minor release? As we support Python 3.7+ this is still ok. 

It would be a rather redundant addition, going through the entire codebase and implementing the appropriate changes, but will enhance readability and keep us better aligned with PEP guidelines.  

This can always be delayed to a later date too, just worth getting some thought's on this.

**Comment by yetinam** [comment]
I agree, it seems like a good idea to replace the type hints in our documentation with proper type hints. Maybe more of a long term project to progressively address? I guess slowly adding type hinting module by module shouldn't break anything.

**Comment by jawooll** [comment]
Definitely, doing it in the way you suggest will break this up into more manageable chunks - sounds good 👍
